lua/starfall/libs_sh/holograms.lua

```diff
@@ -189,6 +189,22 @@ function holograms_library.create(pos, ang, model, scale)
 	end
 end
 
+--- Removes all holograms created by the calling chip
+-- @return number of holograms removed
+function holograms_library.removeAll()
+	if CLIENT and instance.data.render.isRendering then SF.Throw("Cannot remove while in rendering hook!", 2) end
+	
+	local holos = instance.data.holograms.holos
+	local removedHolos = 0
+	
+	for holo, _ in pairs(holos) do
```

Comment:
Can you put `instance.data.holograms.holos` into the `pairs()` instead of creating a local. Also I don't think the `if CLIENT then checkpermission(instance, holo, "hologram.create") end` is needed.

Reply:
Alright, I added it because the hologram:remove() method on the client checks for it.

Reply:
No prob. That one is there to check prop permission, but since this only removes ones the chip created, it should be fine without.